### PR TITLE
ENH: Add dtypes/converters arguments for pandas.read_xml

### LINKED ISSUE #43567: ENH: Possible to add dtype/converters as arguments for pandas.read_xml() ?
#### Is your feature request related to a problem?

I am using pandas lib to read xml for further processes, however a number of columns with leading ZERO are always converted to numbers, so I lost the original data.

#### Describe the solution you'd like

It would be great to add **dtype/converter** arguments for `pandas.read_xml()` to force pandas to interprete certain columns with given dtype/converters.
Just like similar IO read (read_csv, read_html, etc)

[read_xml](https://pandas.pydata.org/docs/reference/api/pandas.read_xml.html)
[read_csv](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html)

#### API breaking implications

Probably not, this argument could be optional.

#### Describe alternatives you've considered

Write my own code to pull data by each xml nodes, which results in very bad performance.

### PR DESCRIPTION
**Pull request by ParfaitG**
- [X] closes #43567
- [X] tests added / passed
- [X] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [X] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.4.0.rst [hunk 1]
```diff
@@ -184,6 +184,47 @@ representation of :class:`DataFrame` objects (:issue:`4889`).
     df
     df.to_dict(orient='tight')
 
+.. _whatsnew_140.read_xml_dtypes:
```
Comment (jreback) [inline doc/source/whatsnew/v1.4.0.rst:187]: move to 1.5

File: pandas/io/xml.py [hunk 2]
```diff
@@ -67,6 +71,23 @@ class _XMLFrameParser:
     names : list
         Column names for Data Frame of parsed XML data.
 
+    dtype : dict
+        Data type for data or columns. E.g. {{'a': np.float64,
+        'b': np.int32, 'c': 'Int64'}}
+
+        .. versionadded:: 1.4.0
```
Comment (jreback) [inline pandas/io/xml.py:78]: same

File: pandas/io/xml.py [hunk 3]
```diff
@@ -109,6 +130,13 @@ def __init__(
         elems_only: bool,
         attrs_only: bool,
         names: Sequence[str] | None,
+        dtype: DtypeArg | None,
+        converters: dict[str, Callable] | None,
+        parse_dates: bool
```
Comment (jreback) [inline pandas/io/xml.py:135]: hmm I think we have a typing alias for this? e.g. is this what we are doing in csv parsers?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/xml.py (hunk 3 above)
Comment (ParfaitG) [reply pandas/io/xml.py:135]: I did look in `pandas._typing`. For `read_csv`, there is no typing for `converters` and `parse_dates`:

```python
def read_csv(
    filepath_or_buffer: FilePath | ReadCsvBuffer[bytes] | ReadCsvBuffer[str],
    ...
    # General Parsing Configuration
    dtype: DtypeArg | None = None,
    ...
    converters=None,
    ...
    # Datetime Handling
    parse_dates=None,
    ...
):
```

Comment (ParfaitG) [reply pandas/io/xml.py:135]: Same with `read_excel`. Let me know how to handle typing for `read_xml`. Maybe raise a `TYP` issue for future PR?

```python
def read_excel(
    io,
    ...
    dtype: DtypeArg | None = None,
    ...
    converters=None,
    ...
    parse_dates=False,
    ...
)
```

File: doc/source/whatsnew/v1.5.0.rst [hunk 4]
```diff
@@ -85,6 +85,48 @@ Optional libraries below the lowest tested version may still work, but are not c
...
 See :ref:`install.dependencies` and :ref:`install.optional_dependencies` for more.
 
+
+.. _whatsnew_140.read_xml_dtypes:
+
+read_xml now supports ``dtype``, ``converters``, and ``parse_dates``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Similar to other IO methods, :func:`pandas.read_xml` now supports assigning specific dtypes to columns,
+apply converter methods, and parse dates.
```
Comment (jreback) [inline doc/source/whatsnew/v1.5.0.rst:95]: can you add the issue reference here (this PR number if no issue)

File: doc/source/whatsnew/v1.5.0.rst [hunk 5]
```diff
@@ -85,6 +85,48 @@ Optional libraries below the lowest tested version may still work, but are not c
 
 See :ref:`install.dependencies` and :ref:`install.optional_dependencies` for more.
 
+
+.. _whatsnew_140.read_xml_dtypes:
```
Comment (jreback) [inline doc/source/whatsnew/v1.5.0.rst:89]: change to 150

File: pandas/io/xml.py (hunk 3 above)
Comment (jreback) [reply pandas/io/xml.py:135]: can you *add* your alias in _typing and use it (can followup later to use it elsewhere)

**Review by jreback — CHANGES_REQUESTED** [review]
small comments & pls merge master, ping on green-ish

File: pandas/_typing.py [hunk 6]
```diff
@@ -129,6 +129,14 @@
 DtypeArg = Union[Dtype, Dict[Hashable, Dtype]]
 DtypeObj = Union[np.dtype, "ExtensionDtype"]
 
+# converters
```
Comment (ParfaitG) [inline pandas/_typing.py:132]: @jreback, how about these new aliases for `converters` and `parse_dates`? (FYI: `DtypeArg` alias was first added for IO SQL #37546).

Comment (jreback) [reply pandas/_typing.py:132]: sounds good

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ParfaitG failures unrelated

**Pull request merged**